File: Numeros Romanos/ruby/DecimalRomano.rb
class Fixnum

  def em_romano

    # Converte número Decimal para Romano

    unidade = ['', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX']
    dezena  = ['', 'X', 'XX', 'XXX', 'XL', 'L', 'LX', 'LXX', 'LXXX', 'XC']
    centena = ['', 'C', 'CC', 'CCC', 'CD', 'D', 'DC', 'DCC', 'DCCC', 'CM']
    milhar  = ['', 'M', 'MM', 'MMM']

    lista = [unidade, dezena, centena, milhar]

    numero = self.to_s.reverse
    casa_decimal = numero.length
    resultado = ''

    casa_decimal.times do
      casa_decimal = casa_decimal - 1
      resultado = resultado + lista[casa_decimal][numero[casa_decimal].to_i]
    end

    resultado

  end

end

File: Numeros Romanos/python/DecimalRomano.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
class Romanos():
  '''
  Classe para trabalhar com números romanos
  '''

  def __init__(self):
    self.unidade = ['', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX']
    self.dezena  = ['', 'X', 'XX', 'XXX', 'XL', 'L', 'LX', 'LXX', 'LXXX', 'XC']
    self.centena = ['', 'C', 'CC', 'CCC', 'CD', 'D', 'DC', 'DCC', 'DCCC', 'CM']
    self.milhar  = ['', 'M', 'MM', 'MMM']

    self.lista = [self.unidade, self.dezena, self.centena, self.milhar]

  def converte(self, numero):
    '''
    Converte número Decimal para Romano
    '''

    casas = str(numero)[::-1]
    resultado = ''

    for i in range(len(casas)):
      resultado = self.lista[i][int(casas[i])] + resultado

    return resultado